arbos/tx_processor.go

```diff
@@ -366,9 +368,7 @@ func (p *TxProcessor) GasChargingHook(gasRemaining *uint64) error {
 	}
 	posterCost, calldataUnits := p.state.L1PricingState().PosterDataCost(p.msg, poster)
 	if calldataUnits > 0 {
-		if err := p.state.L1PricingState().AddToUnitsSinceUpdate(calldataUnits); err != nil {
-			return err
-		}
+		_ = p.state.L1PricingState().AddToUnitsSinceUpdate(calldataUnits)
```

Comment:
I dislike removing error checks. I understand this is the system burner so it shouldn't error, but I'd at least call `burner.Restrict` on the error

Reply:
Sure, I've changed it to `restrict`. I made a few other changes as you saw to make this hook easier to read, though this is the only one around errors